### PR TITLE
implemented sweep spectrum signature for both radio and drone jammers

### LINKED ISSUE #83: Add sweep spectrum signal to radio jammers that traverse entire frequency span as indicator of jammer
Note to self. Implemented handling to support jammer spectrum signals for both drone and radio jammer. Will sweep the entire spectrum of the type, and thus show a sweep in jammer frequency, and a sweep in entire radio comms frequency span if both is enabled. 

However activating that causes following issues:
1. due to the big increase in frequency span going from 7 to 359, even at incrementSpeed of 50, it still "felt too slow". Although might be fine. 
2. Multiple signals on same object makes the zeus text be on top of each other and be impossible to read. Ideally you would want them spaced apart, or aggregated so its a single line showing array of "signal types" too.

### PR DESCRIPTION
**Pull request by Crowdedlight**
Implements so there is ``sweep_drone`` and ``sweep_radio`` types that will sweep the entire radio spectrum or the entire jammer spectrum. To show tracking of jammers. Fixes #83 

https://github.com/Crowdedlight/Crows-Electronic-Warfare/assets/7889925/c64fdaf0-d304-465c-b013-681b669682d2

**Problem**
Multiple signals on the same object will overlap and make it impossible to read. Need to find a solution that is also fast performing in the onDraw loop. Not quite sure what approach to go. 

billede

### CODE REVIEW

**Comment by b-mayr-1984** [comment]
> **Problem** Multiple signals on the same object will overlap and make it impossible to read. Need to find a solution that is also fast performing in the onDraw loop. Not quite sure what approach to go.
> 
> billede

You have the information to which `_unit` a beacon belongs to available [here](https://github.com/Crowdedlight/Crows-Electronic-Warfare/blob/sweep_fix/addons/zeus/functions/fnc_addZeusTextDisplayEH.sqf#L39).
You could use [setVariable](https://community.bistudio.com/wiki/setVariable) on that object to know what has already been drawn.

If you really care about performance I recommend this approach in general. Should be faster to just let objects know what information need to be put into a 3D text rather than running through similar `forEach` loops (all with their own copy of `if (_dist > 500) then {continue;}` which is called multiple times for the same object). Is a bit of a re-write though.

File: addons/spectrum/functions/fnc_spectrumDeviceMouseDown.sqf [hunk 1]
```diff
@@ -125,7 +125,8 @@ if (count _frequenciesSorted > 0) then {
 			_radio = true;
 			_timeActive = 0.1;
 		};
-		case "sweep";	// Players can not interact with a drone jammer. So just ignore this signal.
+		case "sweep_drone";	// Players can not interact with a drone jammer. So just ignore this signal.
+		case "sweep_radio";	// Players can not interact with a drone jammer. So just ignore this signal.
```
Comment (b-mayr-1984) [inline addons/spectrum/functions/fnc_spectrumDeviceMouseDown.sqf:129]: Copy'n'pasted comment does not really fit `"sweep_radio"`.

Comment (Crowdedlight) [reply addons/spectrum/functions/fnc_spectrumDeviceMouseDown.sqf:129]: Good catch

**Comment by Crowdedlight** [comment]
> > **Problem** Multiple signals on the same object will overlap and make it impossible to read. Need to find a solution that is also fast performing in the onDraw loop. Not quite sure what approach to go.
> > billede
> 
> You have the information to which `_unit` a beacon belongs to available [here](https://github.com/Crowdedlight/Crows-Electronic-Warfare/blob/sweep_fix/addons/zeus/functions/fnc_addZeusTextDisplayEH.sqf#L39). You could use [setVariable](https://community.bistudio.com/wiki/setVariable) on that object to know what has already been drawn.
> 
> If you really care about performance I recommend this approach in general. Should be faster to just let objects know what information need to be put into a 3D text rather than running through similar `forEach` loops (all with their own copy of `if (_dist > 500) then {continue;}` which is called multiple times for the same object). Is a bit of a re-write though.

Yeah, I had played with the idea of putting the info on the object, maybe at a start just for beacons. I would still need an if-statement as its used to hide the text when zeus is not close in zeus view. I seem to recall I had issues that if I didn't do that the text would be visible when far away even when you can't see the object itself. The text doesn't scale and fade out by itself like engine zeus icons. Or at least didn't when I tested it back then.

**Comment by Crowdedlight** [comment]
billede

**Pull request merged**